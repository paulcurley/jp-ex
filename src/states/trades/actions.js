export const ADD_TRADE = 'ADD_TRADE';

export const addTrade = ({ trade }) => ({
    type: ADD_TRADE,
    trade,
});
