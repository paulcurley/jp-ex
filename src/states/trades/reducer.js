import { ADD_TRADE } from './actions';

const defaultState = [];

const reducer = (state = defaultState, action) => {
    switch (action.type) {
    case ADD_TRADE:
        return [
            {
                timestamp: Date.now(),
                ...action.trade,
            },
            ...state,
        ];
    default:
        return state;
    }
};

export default reducer;
