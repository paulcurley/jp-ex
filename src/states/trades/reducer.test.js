import reducer from './reducer';
import { ADD_TRADE } from './actions';

Date.now = jest.fn(() => '11111111');
describe('trade Reducer', () => {
    it('ADD_TRADE should update mode', () => {
        const initialState = {};

        const action = {
            type: ADD_TRADE,
            trade: {
                stockSymbol: 'ALC',
                price: 43543,
                noOfShares: 543,
            },
        };

        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
    });
});
