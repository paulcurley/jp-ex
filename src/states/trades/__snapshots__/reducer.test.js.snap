// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trade Reducer ADD_TRADE should update mode 1`] = `
Array [
  Object {
    "noOfShares": 543,
    "price": 43543,
    "stockSymbol": "ALC",
    "timestamp": "11111111",
  },
]
`;
